theano/scan_module/scan_checkpoint.py

```diff
@@ -75,7 +75,7 @@ def outer_step(*args):
                                   n_steps=o_n_steps, allow_gc=True)
                                   
     # Keep only the last timestep of every output but keep all the updates
-    return results, updates
+    return results, updates # TODO is it a bug?
```

Comment:
I think we should remove this line. @carriepl do you remember why you added it? As for this to work well, we the user to only use the last step. So to prevent problems, I would think to not use the same interface as Scan. This would help prevent user from using all the states information (and so not get any memory saving).

@carriepl what do you think?